packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,406 @@
+      {description && <p className="mt-2 text-xs">{description}</p>}
+    </div>
+  )
+}
+
+function NetworkRow({
+  chainId,
+  style,
+  onClick,
+  close
+}: {
+  chainId: ChainId
+  style: CSSProperties
+  onClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const { network, nativeTokenData } = getBridgeUiConfigForChain(chainId)
+  const chain = getWagmiChain(chainId)
+  const [{ sourceChain }] = useNetworks()
+
+  function handleClick() {
+    onClick(chain)
+    close() // close the popover after option-click
+  }
+
+  function onKeyDown(event: React.KeyboardEvent<HTMLButtonElement>) {
+    if (event.key === 'Enter') {
+      handleClick()
+    }
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      onKeyDown={onKeyDown}
+      key={chainId}
+      style={style}
+      type="button"
+      aria-label={`Switch to ${network.name}`}
+      className={twMerge(
+        'flex h-[90px] w-full items-center gap-4 px-6 py-2 text-lg hover:bg-black/10',
+        chainId === sourceChain.id && 'bg-black/10' // selected row
+      )}
+    >
+      <span className="flex h-6 w-6 shrink-0 items-center justify-center lg:h-6 lg:w-6">
+        <Image
+          src={network.logo}
+          alt={`${network.name} logo`}
+          className="h-full w-auto"
+          width={24}
+          height={24}
+        />
+      </span>
+      <div className={twMerge('flex flex-col items-start gap-1')}>
+        <span className="truncate leading-none">{network.name}</span>
+        {network.description && (
+          <p className="whitespace-pre-wrap text-left text-xs leading-[1.15]">
+            {network.description}
+          </p>
+        )}
+        {network.type && (
+          <p className="text-[10px] leading-none">
+            {network.type} Chain, {nativeTokenData?.symbol ?? 'ETH'} is the
+            native gas token
+          </p>
+        )}
+      </div>
+    </button>
+  )
+}
+
+function AddCustomOrbitChainButton() {
+  const [, setQueryParams] = useArbQueryParams()
+  const [isTestnetMode] = useIsTestnetMode()
+
+  if (!isTestnetMode) {
+    return null
+  }
+
+  return (
+    <button
+      className="text-sm underline"
+      onClick={() => setQueryParams({ settingsOpen: true })}
+    >
+      <span>Add Custom Orbit Chain</span>
+    </button>
+  )
+}
+
+function NetworksPanel({
+  chainIds,
+  onNetworkRowClick,
+  close
+}: {
+  chainIds: ChainId[]
+  onNetworkRowClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const [errorMessage, setErrorMessage] = useState('')
+  const [networkSearched, setNetworkSearched] = useState('')
+  const [debouncedNetworkSearched, setDebouncedNetworkSearched] = useState('')
+  const listRef = useRef<List>(null)
+  const [isTestnetMode] = useIsTestnetMode()
+
+  const testnetToggleClassNames = {
+    switch:
+      'ui-checked:bg-black/20 ui-not-checked:bg-black/20 [&_span]:ui-not-checked:bg-black'
+  }
+
+  useDebounce(
```

Comment:
We also have `useDebouncedValue` which I think is a bit more intuitive to use

Reply:
i've changed it to https://usehooks.com/usedebounce instead